libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/RoutesRenderedCallback.kt

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.navigation.ui.maps.route.line.api
+
+import com.mapbox.maps.MapboxMap
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+
+/**
+ Callback to be notified of rendered routes.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+interface RoutesRenderedCallback {
+
+    /**
+     * Map instance where the routes are expected to be rendered.
+     */
+    val map: MapboxMap
+
+    /**
+     * Invoked when routes are rendered.
+     *
+     * @param result [RoutesRenderedResult] describing rendering result
+     */
+    fun onRoutesRendered(result: RoutesRenderedResult)
+}
+
+/**
+ * Class describing the result of routes rendering.
+ *
+ * @param successfullyRenderedRouteIds ids of the routes that were successfully rendered (correspond to [NavigationRoute.id])
+ * @param renderingCancelledRouteIds ids of the routes whose rendering was cancelled (correspond to [NavigationRoute.id]).
+ *  This may happen, for example, when a newer rendering operation had been queued before the previous one finished.
+ * @param successfullyClearedRouteIds ids of the routes that were successfully cleared (correspond to [NavigationRoute.id])
+ * @param clearingCancelledRouteIds ids of the routes whose clearance was cancelled (correspond to [NavigationRoute.id]).
+ *  This may happen, for example, when a newer rendering operation had been queued before the previous one finished.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class RoutesRenderedResult internal constructor(
+    val successfullyRenderedRouteIds: Set<String>,
```

Comment:
We call the callback for a group of routes, while under the hood they are drown one by one. It can cause the opposite effect: route objects won't be drown until all 3 routes are visible.

Reply:
Yes, but the alternative is to have callback be invoked multiple number of times, which doesn't fit into callback pattern.
The observer pattern was discarded earlier, more context: https://github.com/mapbox/mapbox-navigation-android/pull/6964#issuecomment-1441621153
And a technical limitation: https://github.com/mapbox/mapbox-navigation-android/pull/6964#discussion_r1116185159